src/app/pages/home/activity/transaction-details/transaction-details.page.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { UntilDestroy } from '@ngneat/until-destroy';
+import { map } from 'rxjs/operators';
+import { Transaction } from '../../../../services/publisher/numbers-storage/numbers-storage-api.service';
+
+@UntilDestroy({ checkProperties: true })
+@Component({
+  selector: 'app-transaction-details',
+  templateUrl: './transaction-details.page.html',
+  styleUrls: ['./transaction-details.page.scss'],
+})
+export class TransactionDetailsPage implements OnInit {
+  details!: Transaction;
+  readonly status = Status;
+
+  constructor(
+    private readonly router: Router,
+    private readonly route: ActivatedRoute
+  ) {}
+
+  ngOnInit() {
+    this.route.paramMap
+      .pipe(map(() => window.history.state))
+      .subscribe(details => {
+        this.details = details;
```

Comment:
Set `this.details` directly in its declaration by using `Observable`. For example,

```typescript
details$ = this.route.paramMap.pipe(
  map(() => history.state)
)
```